twill-yarn/src/main/java/org/apache/twill/internal/ServiceMain.java

```diff
@@ -145,9 +145,8 @@ protected static Location createAppLocation(final Configuration conf) {
       if (UserGroupInformation.isSecurityEnabled()) {
         ugi = UserGroupInformation.getCurrentUser();
       } else {
-        String fsUser = System.getenv(EnvKeys.TWILL_FS_USER);
         if (fsUser == null) {
-          throw new IllegalStateException("Missing environment variable " + EnvKeys.TWILL_FS_USER);
+          throw new IllegalStateException("Missing environment variable " + Constants.TWILL_FS_USER);
```

Comment:
This error message is not right. The fs user now comes in the runtime spec, not from env, right? In fact, can this happen?